BLUESPAWN-client/src/util/eventlogs/EventLogs.cpp

```diff
@@ -211,12 +188,54 @@ DWORD ProcessResults(EVT_HANDLE hResults, Reaction& reaction, int* numFound, std
 	return status;
 }
 
-int QueryEvents(const wchar_t* channel, unsigned int id, Reaction& reaction)
+DWORD EventLogs::EventToDetection(EVT_HANDLE hEvent, EVENT_DETECTION* pDetection, std::set<std::wstring>& params) {
+	DWORD status = ERROR_SUCCESS;
+
+	std::wstring eventIDStr;
+	std::wstring eventRecordIDStr;
+	std::wstring timeCreated;
+	std::wstring channel;
+	std::wstring rawXML;
+
+	if (ERROR_SUCCESS != (status = GetEventParam(hEvent, &eventIDStr, L"Event/System/EventID")))
+		return status;
+	if (ERROR_SUCCESS != (status = GetEventParam(hEvent, &eventRecordIDStr, L"Event/System/EventRecordID")))
+		return status;
+	if (ERROR_SUCCESS != (status = GetEventParam(hEvent, &timeCreated, L"Event/System/TimeCreated/@SystemTime")))
+		return status;
+	if (ERROR_SUCCESS != (status = GetEventParam(hEvent, &channel, L"Event/System/Channel")))
+		return status;
+	if (ERROR_SUCCESS != (status = GetEventXML(hEvent, &rawXML)))
+		return status;
+
+	// Specify extra parameters
+	std::unordered_map<std::wstring, std::wstring> extraParams;
+	for (std::wstring key : params) {
+		std::wstring val;
+		if (ERROR_SUCCESS != (status = GetEventParam(hEvent, &val, key))) {
+			LOG_ERROR(L"EventLogs::EventToDetection: Failed query parameter " + key + L" with code " + std::to_wstring(status));
+			return status;
+		}
+
+		extraParams.insert({ key, val });
+	}
+
+	pDetection->eventID = std::stoul(eventIDStr);
+	pDetection->eventRecordID = std::stoul(eventRecordIDStr);
+	pDetection->timeCreated = timeCreated;
+	pDetection->timeCreated = channel;
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Variable 'pDetection->timeCreated' is reassigned a value before the old one has been used.](https://app.codacy.com/manual/ION28/BLUESPAWN/pullRequest?prid=4960363)